Source/sha.cpp

```diff
@@ -50,9 +50,9 @@ void SHA1Input(SHA1Context *context, const char *message_array, int len)
 
 void SHA1ProcessMessageBlock(SHA1Context *context)
 {
-	int i, temp;
-	int W[80];
-	int A, B, C, D, E;
+	DWORD i, temp;
+	DWORD W[80];
+	DWORD A, B, C, D, E; // Only works with `int`, perhaps it relies on sign-extension???

```

Comment:
@AJenbo Are our save files compatible with original Diablo?
If so, then the SHA1 implementation in OG Diablo is wrong :sweat_smile:

Reply:
look at asm? :p